Pokemon/Pokemon/PokemonTableViewController.m

```diff
@@ -0,0 +1,506 @@
+                                  @"Nidoking",
+                                  @"Diglett",
+                                  @"Dugtrio",
+                                  @"Geodude",
+                                  @"Graveler",
+                                  @"Golem",
+                                  @"Onix",
+                                  @"Cubone",
+                                  @"Marowak",
+                                  @"Rhyhorn",
+                                  @"Rhydon"],
+                     @"Fairy": @[@"Clefairy",
+                                 @"Clefable",
+                                 @"Jigglypuff",
+                                 @"Wigglytuff",
+                                 @"Mr. Mime"],
+                     @"Fighting": @[@"Mankey",
+                                    @"Primeape",
+                                    @"Poliwrath",
+                                    @"Machop",
+                                    @"Machoke",
+                                    @"Machamp",
+                                    @"Hitmonlee",
+                                    @"Hitmonchan"],
+                     @"Psychic": @[@"Abra",
+                                   @"Kadabra",
+                                   @"Alakazam",
+                                   @"Slowpoke",
+                                   @"Slowbro",
+                                   @"Drowzee",
+                                   @"Hypno",
+                                   @"Exeggcute",
+                                   @"Exeggutor",
+                                   @"Starmie",
+                                   @"Mr. Mime",
+                                   @"Jynx",
+                                   @"Mewtwo",
+                                   @"Mew"],
+                     @"Rock": @[@"Geodude",
+                                @"Graveler",
+                                @"Golem",
+                                @"Onix",
+                                @"Rhyhorn",
+                                @"Rhydon",
+                                @"Omanyte",
+                                @"Omastar",
+                                @"Kabuto",
+                                @"Kabutops",
+                                @"Aerodactyl"],
+                     @"Steel": @[@"Magnemite",
+                                 @"Magneton"],
+                     @"Ice": @[@"Dewgong",
+                               @"Cloyster",
+                               @"Jynx",
+                               @"Lapras",
+                               @"Articuno"],
+                     @"Ghost": @[@"Gastly",
+                                 @"Haunter",
+                                 @"Gengar"],
+                     @"Dragon": @[@"Dratini",
+                                  @"Dragonair",
+                                  @"Dragonite"]
+                                  };
+    self.pokemonArray = @[@"Squirtle",
```

Comment:
You could derive this list from the above (separated) lists, or find some way to represent all the data in one format and then derive the A-Z and type-separated lists from that format. Generally it's a good idea to avoid duplicating data; this is sometimes called DRY or the DRY principle ("Don't Repeat Yourself").

http://c2.com/cgi/wiki?DontRepeatYourself